chpl/chpl-service/src/main/java/gov/healthit/chpl/web/controller/MeaningfulUseController.java

```diff
@@ -0,0 +1,185 @@
+					try{
+						numMeaningfulUseUsers = Long.parseLong(currRecord.get(1).trim());
+						muu.setProductNumber(chplProductNumber);
+						muu.setNumberOfUsers(numMeaningfulUseUsers);
+						muu.setCsvLineNumber(i);
+						// check if product number has already been updated
+						if(uniqueMuusFromFile.contains(muu.getProductNumber())){
+							throw new IOException();
+						}
+						muusToUpdate.add(muu);
+						uniqueMuusFromFile.add(muu.getProductNumber());
+					} catch (NumberFormatException e){
+						muu.setProductNumber(chplProductNumber);
+						muu.setCsvLineNumber(i);
+						muu.setError("Line " + muu.getCsvLineNumber() + ": Field \"num_meaningful_use\" with value \"" + currRecord.get(1).trim() + "\" is invalid. "
+								+ "Value in field \"num_meaningful_use\" must be an integer.");
+						muusToUpdate.add(muu);
+						uniqueMuusFromFile.add(muu.getProductNumber());
+					}
+					catch (IOException e){
+						muu.setProductNumber(chplProductNumber);
+						muu.setCsvLineNumber(i);
+						Integer dupLineNumber = null;
+						// get line number with duplicate chpl_product_number
+						for (MeaningfulUseUser entry: muusToUpdate) {
+						     if (entry.getProductNumber().equals(muu.getProductNumber())){
+						    	 dupLineNumber = entry.getCsvLineNumber();
+						     }
+						   }
+						muu.setError("Line " + muu.getCsvLineNumber() + ": Field \"chpl_product_number\" with value \"" + muu.getProductNumber() + "\" is invalid. "
+								+ "Duplicate \"chpl_product_number\" at line " + dupLineNumber);
+						muusToUpdate.add(muu);
+					}
+				}
+			}
+		} catch(IOException ioEx) {
+			logger.error("Could not get input stream for uploaded file " + file.getName());			
+			throw new ValidationException("Could not get input stream for uploaded file " + file.getName());
+		} finally {
+			 try { parser.close(); } catch(Exception ignore) {}
+			try { reader.close(); } catch(Exception ignore) {}
+		}
+		
+		try {
+			meaningfulUseUserResults = cpManager.updateMeaningfulUseUsers(muusToUpdate);
+		} catch (JsonProcessingException e) {
+			e.printStackTrace();
+		} catch (EntityCreationException e) {
+			e.printStackTrace();
+		} catch (EntityRetrievalException e) {
+			e.printStackTrace();
+		} catch (IOException e) {
+			e.printStackTrace();
+		} 
+		
+		return meaningfulUseUserResults;
+	}
+	
+	@ApiOperation(value="Get a single date value to indicate when the last meaningful use user file data is accurate as of.", 
+			notes="This is a single system-wide value.")
+	@RequestMapping(value="/accurate_as_of", method=RequestMethod.GET,
+			produces="application/json; charset=utf-8") 
+	public @ResponseBody MeaningfulUseAccurateAsOf getAccurateAsOfDate() {
+		MeaningfulUseAccurateAsOfDTO dto = muManager.getMeaningfulUseAccurateAsOf();
+		MeaningfulUseAccurateAsOf muuAccurate = new MeaningfulUseAccurateAsOf(dto);
+		if(StringUtils.isEmpty(muuAccurate.getId().toString()) && StringUtils.isEmpty(muuAccurate.getAccurateAsOfDate().toString())){
```

Comment:
- If muuAccurate.getId() is null or muuAccurate.getAccurateAsOfDate() is null this will throw an NPE. 

- Unless there are requirements for other reasons to use errorMessages and warningMessages, I don't think that is appropriate here since there is only ever one error message and never any warning messages. You probably just want to throw an exception of a type that is handled as Controller Advice (or add a handler for a different type to that class) with the error message. Maybe EntityNotFoundException? or EntityExistsException?